### PR TITLE
Add a sync_burst_spread option

### PR DESCRIPTION
**Pull request by oserres**
Add a new sync_burst_spread option in which all the service instances start the burst at a different time.

### CODE REVIEW

File: distbench_engine.cc [hunk 1]
```diff
@@ -886,11 +886,20 @@ void DistBenchEngine::RunAction(ActionState* action_state) {
     CHECK_EQ(action_state->next_iteration_time, absl::InfiniteFuture());
     absl::Duration period = absl::Nanoseconds(
         action_state->action->proto.iterations().open_loop_interval_ns());
-    if (action_state->action->proto.iterations()
-        .open_loop_interval_distribution() == "sync_burst") {
+    auto interval_distribution = action_state->action->proto.iterations()
```
Comment (danmanj) [inline distbench_engine.cc:889]: auto&

File: distbench_engine.cc [hunk 2]
```diff
@@ -886,11 +886,20 @@ void DistBenchEngine::RunAction(ActionState* action_state) {
...
+        .open_loop_interval_distribution();
+    if (interval_distribution == "sync_burst") {
       absl::Duration start = clock_->Now() - absl::UnixEpoch();
       action_state->next_iteration_time =
         period + absl::UnixEpoch() + absl::Floor(start, period);
+    } else if (interval_distribution == "sync_burst_spread") {
+      absl::Duration start = clock_->Now() - absl::UnixEpoch();
+      int nb_peers = peers_[action_state->rpc_service_index].size();
```
Comment (danmanj) [inline distbench_engine.cc:897]: double

File: distbench_engine.cc [hunk 3]
```diff
@@ -886,11 +886,20 @@ void DistBenchEngine::RunAction(ActionState* action_state) {
...
+      double fraction = (double)service_instance_ / nb_peers;
```
Comment (danmanj) [inline distbench_engine.cc:898]: no need for c-style cast if var is double to begin with.

File: distbench_engine.cc (hunk 1 above)
Comment (oserres) [reply distbench_engine.cc]: Done

File: distbench_engine.cc (hunk 3 above)
Comment (oserres) [reply distbench_engine.cc]: Done but I think it was clearer with the c-style cast...

**Review by danmanj — APPROVED** [review]

**Pull request merged**